privacyidea/api/lib/prepolicy.py

```diff
@@ -1161,6 +1161,16 @@ def check_base_action(request=None, action=None, anonymous=False):
             realm = get_realms_of_token(request.view_args.get("serial"),
                                         only_first_realm=True)
 
+        # if the resolver was not given in the request,
+        # get the resolver by the serial either from request or from view_args
+        if not resolver:
```

Comment:
I think the resolver is already available from 1138.

Or - We could make the resolver present here by adding the resolver as a return value in `determine_logged_in_userparams` in line 1139.